src/database/utils.ts

```diff
@@ -69,3 +70,14 @@ export function buildSearchCollectionsWhereClause(
 
   return where;
 }
+
+export function checkCollectionLabelLimit(labelExternalIds: string[]) {
+  if (labelExternalIds.length > config.app.collectionLabelLimit) {
+    const labelsToRemove =
+      labelExternalIds.length - config.app.collectionLabelLimit;
+    const labelsSingularOrPlural = labelsToRemove === 1 ? 'label' : 'labels';
+    throw new UserInputError(
+      `Too many labels provided. Max is ${config.app.collectionLabelLimit}. Please remove ${labelsToRemove} ${labelsSingularOrPlural}.`
```

Comment:
I'm on the fence about the wording of this error message. It's on the long side and specifies the number of labels to remove as a number, not a word, which is awkward to read - there's an example in the tests. 

One way to improve the existing wording is to use `toWords()` to specify the number of labels to remove. 

Maybe the wording of the entire message could be reworked, I'm not sure. This is an edge case that will probably not be seen that often in real life, so I'm trying not to go overboard with suggestions for perfect wording. How about something along the lines of "Too many labels: 21 out of 20 max provided"? This would also eliminate the stylistic need for converting numbers into words! 🙃

Reply:
@nina-py mmm yes, I was thinking how to word this one. I do want to definitely provide the max allowed # and the users input in the message. I do think it would be useful to provide how many labels to remove --- say our max is 20 and someone goes overboard and uses 36 labels, it would help to show that they should remove 16 labels. Idk tbh. 

In this kind of use case it also makes me think we should have the functionality to remove the extra labels:
* User enters too many labels
* We show error message and option "remove N # of labels?" and if user clicks that, it would remove the last N extra labels. 

I see this being useful in cases where user has more than 5 extra labels for example. This would save time for them rather than clicking N # of times to delete the extra labels. 

Anyways this is getting out of the scope for this ticket, maybe we can reword the message to: "The number of provided labels must not be more than 20. It is currently 21".

Reply:
I think removing extra labels is fraught with danger as we wouldn't necessarily be removing the labels the user would have chosen to remove if there was a need to pare the number of labels down. The user would probably prioritise some labels over others instead of chopping off the end of the list which we'd do programmatically.

I really like the last wording you provided, I think it's the best of the lot. Short and has all the actionable information.